src/global.rs

```diff
@@ -10,15 +10,34 @@ use chacha20poly1305::XChaCha20Poly1305;
 pub const BLOCK_SIZE: usize = 1_048_576; // 1024*1024 bytes
 pub const SALT_LEN: usize = 16; // bytes
 
-#[allow(clippy::struct_excessive_bools)] // possibly only temporary - requires more thought
 pub struct Parameters {
-    pub hash_mode: bool,
-    pub skip: bool,
-    pub bench: bool,
-    pub password: bool,
+    pub hash_mode: HashMode,
+    pub skip: SkipMode,
+    pub bench: BenchMode,
+    pub password: PasswordMode,
     pub cipher_type: CipherType,
 }
+#[derive(PartialEq, Eq, Clone, Copy)]
+pub enum HashMode {
```

Comment:
I feel as though these types could be `CalculateHash` and `NoHash` (or something similar). In `HideHash` mode, the hash isn't calculated but the name suggests otherwise.

No worries if you disagree - I couldn't really think of suitable names for these types anyway :)